File: OPENAPI-REST-API/openapi-client/zapier/models/LicenseOrderRequest.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}package`,
                label: `License service type id from getNewLicense.serviceTypes (must be buyable). - [${labelPrefix}package]`,
                required: true,
                type: 'integer',
            },
            {
                key: `${keyPrefix}ip`,
                label: `IP address the license is bound to. - [${labelPrefix}ip]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}tos`,
                label: `Terms-of-service acceptance. Must be true to place the order. - [${labelPrefix}tos]`,
                required: true,
                type: 'boolean',
            },
            {
                key: `${keyPrefix}frequency`,
                label: `Billing frequency in months. - [${labelPrefix}frequency]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}coupon`,
                label: `Coupon code. - [${labelPrefix}coupon]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}comment`,
                label: `Free-form note saved on the service row. - [${labelPrefix}comment]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'package': bundle.inputData?.[`${keyPrefix}package`],
            'ip': bundle.inputData?.[`${keyPrefix}ip`],
            'tos': bundle.inputData?.[`${keyPrefix}tos`],
            'frequency': bundle.inputData?.[`${keyPrefix}frequency`],
            'coupon': bundle.inputData?.[`${keyPrefix}coupon`],
            'comment': bundle.inputData?.[`${keyPrefix}comment`],
        }
    },
}

File: OPENAPI-REST-API/openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/LicenseOrderRequest.scala

package org.openapitools.client.model


case class LicenseOrderRequest (
    /* License service type id from getNewLicense.serviceTypes (must be buyable). */
    _package: Integer,
    /* IP address the license is bound to. */
    _ip: String,
    /* Terms-of-service acceptance. Must be true to place the order. */
    _tos: Boolean,
    /* Billing frequency in months. */
    _frequency: Option[Integer],
    /* Coupon code. */
    _coupon: Option[String],
    /* Free-form note saved on the service row. */
    _comment: Option[String]
)
object LicenseOrderRequest {
    def toStringBody(var_package: Object, var_ip: Object, var_tos: Object, var_frequency: Object, var_coupon: Object, var_comment: Object) =
        s"""
        | {
        | "package":$var_package,"ip":$var_ip,"tos":$var_tos,"frequency":$var_frequency,"coupon":$var_coupon,"comment":$var_comment
        | }
        """.stripMargin
}
